app/src/main/java/io/apicurio/registry/storage/impl/sql/CommonSqlStatements.java

```diff
@@ -574,6 +602,11 @@ public String selectContentByContentHash() {
                 + "WHERE v.tenantId = ? AND c.contentHash = ?";
     }
 
+    @Override
+    public String deleteAllOrphanedContent() {
+        return "DELETE FROM content c WHERE c.contentId NOT IN (SELECT v.contentId FROM versions v)";
```

Comment:
this could be problematic if the number of rows returned by the subquery is too large. I think in this situations a safer solution is to get the list of ids with a select query, slice it and run the delete query with each slice of ids.
The same concert applies to other similar queries above. However I think this is the most critical one, because this query is not using tenantId to filter any data and that makes it the most problematic

Reply:
I'll restrict this by a tenantId, good point. I don't expect the # of rows will be too large, since this will also be called in deleteArtifact methods in `AbstractSqlRegistryStorage` (although I think I missed some).